pkg/registry/apis/dashboards/access/types.go

```diff
@@ -0,0 +1,44 @@
+package access
+
+import (
+	"context"
+
+	"github.com/grafana/grafana/pkg/apis/dashboards/v0alpha1"
+)
+
+type DashboardRow struct {
+	// Dashboard resource
+	Dash *v0alpha1.Dashboard
+
+	// Title -- this may come from saved metadata rather than the body
+	Title string
+
+	// The folder UID (needed for access control checks)
+	FolderUID string
+
+	// Needed for fast summary access
+	Tags []string
+
+	// Size (in bytes) of the dashboard payload
+	Bytes int
+
+	// Last time the dashboard was updated (used in the continue token)
+	UpdatedTime int64
+
+	// The token we can use that will start a new connection that includes
+	// this same dashboard
+	ContinueToken string
+}
+
+type DashboardRows interface {
+	Close() error
+	Next() (*DashboardRow, error)
+}
+
+// This does not check if you have permissions!
+type DashboardAccess interface {
+	GetDashboard(ctx context.Context, orgId int64, uid string) (*DashboardRow, error)
+	GetDashboards(ctx context.Context, orgId int64, continueToken string, skipBody bool) (DashboardRows, error)
```

Comment:
having slept on it... maybe better to return `v0alpha1.Dashboard` and `v0alpha1.DashboardList` directly, but take in a max bytes argument.

then we don't have to expose the potential Close leak 🤔

Reply:
I have had success in the past streaming very large result sets all the way to the http response, but unless you are encoding and flushing each item separately you may as well accumulate the list here vs taking the complexity just to accumulate it in memory higher up the stack before passing the whole thing to the encoder.

Reply:
for this case, i think you are right, with k8s we have to let it decide the JSON vs Yaml format anyway (without MASSIVE effort)